### PR TITLE
adding build, unit test, style and verify github actions

### PR DESCRIPTION
**Pull request by n3wscott**
These will run with each PR, but until we require them, they will be "just for fun".

### CODE REVIEW

**Comment by n3wscott** [comment]
The actions will not run off a PR from a fork. The will have to land and then they run on the next PRs.

File: .github/workflows/go-pr-build.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+          path: ./src/knative.dev/${{github.event.pull_request.head.repo.name}}
+
+      - name: Vet
+        run: go vet knative.dev/${{github.event.pull_request.head.repo.name}}/...
+
+      - name: Build
+        run: go build -v knative.dev/${{github.event.pull_request.head.repo.name}}/...
```
Comment (mattmoor) [inline .github/workflows/go-pr-build.yaml]: add newlines

File: .github/workflows/go-pr-verify.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    - name: Install Dependencies
+      run: |
+        go get github.com/google/ko/cmd/ko
+        go get github.com/otiai10/copy
+        go get github.com/google/go-licenses
+
+    - name: Check out code onto GOPATH
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
```
Comment (mattmoor) [inline .github/workflows/go-pr-verify.yaml:38]: why 2?

File: .github/workflows/go-pr-verify.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    - name: Set up Go ${{ matrix.go-version }}
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version }}
+      id: go
+
+    - name: Install Dependencies
+      run: |
+        go get github.com/google/ko/cmd/ko
+        go get github.com/otiai10/copy
```
Comment (mattmoor) [inline .github/workflows/go-pr-verify.yaml:32]: what's this for?

File: .github/workflows/go-pr-verify.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+name: Verify
```
Comment (mattmoor) [inline .github/workflows/go-pr-verify.yaml:1]: copyright headers

File: .github/workflows/go-pr-style.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          go-version: 1.14.x
+        id: go
+
+      - name: Check out code onto GOPATH
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+          path: ./src/knative.dev/${{github.event.pull_request.head.repo.name}}
+
+      # TODO: add pretter step
```
Comment (vagababov) [inline .github/workflows/go-pr-style.yaml:30]:
```suggestion
      # TODO: add prettier step
```

File: .github/workflows/go-pr-verify.yaml (hunk 3 above)
Comment (n3wscott) [reply .github/workflows/go-pr-verify.yaml]: dropped.

**Comment by n3wscott** [comment]
omg come on prow... why you take 15 minutes for pkg-build-test?!!

**Comment by n3wscott** [comment]
/retest
flake.

**Comment by vagababov** [comment]
I wonder if it should have applied to itself? Or did it 🤯

**Comment by vagababov** [comment]
@n3wscott it seem we need to add go-cov?

**Pull request merged**